source/interact-data/scoping.txt

```diff
@@ -0,0 +1,260 @@
+association that affects its visibility when the scope is applied.
+
+This example uses the following models:
+
+.. literalinclude:: /includes/interact-data/scoping.rb
+   :start-after: start-scope-association
+   :end-before: end-scope-association
+   :language: ruby
+   :dedent:
+
+Suppose you create a ``Label`` model that contains an association to a
+``Band`` in which the value of ``active`` is ``true``. When you update
+the ``active`` field to ``false``, {+odm+} still loads it despite the
+default scope. To view the documents in the association with the scope
+applied, you must call the ``reload()`` operator.
+
+The following code demonstrates this sequence:
+
+.. literalinclude:: /includes/interact-data/scoping.rb
+   :start-after: start-scope-association-steps
+   :end-before: end-scope-association-steps
+   :language: ruby
+   :dedent:
+   :emphasize-lines: 5, 7
+
+or and nor Query Behavior
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+{+odm+} treats the criteria in a default scope the same way as any other
+query conditions. This can lead to surprising behavior when using the
+``or()`` and ``nor()`` methods.
+
+The following examples demonstrate how {+odm+} interprets queries on
+models with a default scope:
+
+.. literalinclude:: /includes/interact-data/scoping.rb
+   :start-after: start-scope-query-behavior
+   :end-before: end-scope-query-behavior
+   :language: ruby
+   :dedent:
+
+To learn more about logical operations, see
+:ref:`mongoid-query-logical-operations` in the Specify a Query guide.
+
+Disable Scope When Querying
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can direct {+odm+} to not apply the default scope by using the
+``unscoped`` operator, as shown in the following examples:
+
+.. literalinclude:: /includes/interact-data/scoping.rb
+   :start-after: start-unscoped
+   :end-before: end-unscoped
+   :language: ruby
+   :dedent:
+
+Override Default Scope at Runtime
+---------------------------------
+
+You can use the ``with_scope()`` method to change the default scope in a
+block at runtime.
+
+The following model defines the *named* scope ``mexican``:
+
+.. literalinclude:: /includes/interact-data/scoping.rb
+   :start-after: start-override-scope
+   :end-before: end-override-scope
+   :language: ruby
+   :dedent:
+   :emphasize-lines: 7
+
+You can use the ``with_scope()`` method to set the ``mexican`` named
+scope as the default scope at runtime, as shown in the following code:
+
+.. literalinclude:: /includes/interact-data/scoping.rb
+   :start-after: start-override-scope-block
+   :end-before: end-override-scope-block
+   :language: ruby
+   :dedent:
+
+Class Methods
+-------------
+
+{+odm+} treats class methods on models that return ``Criteria`` objects
+as scopes. You can chain these class methods when querying, as shown in
+the following example:
+
+.. literalinclude:: /includes/interact-data/scoping.rb
+   :start-after: start-class-methods
+   :end-before: end-class-methods
+   :language: ruby
+   :dedent:
+   :emphasize-lines: 7-9, 12
```

Comment:
Q (again, maybe unfamiliarity with syntax): what methods are being chained in this example? I guess "chained" makes me imagine multiple class methods being used.

Reply:
I can reword! I guess in this example a method is being chained to Band but really its the first method.